pages/api/program/[id].ts

```diff
@@ -0,0 +1,54 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { ResponseUtil } from "@utils/responseUtil";
+import { getProgram, deleteProgram } from "@database/program";
+
+/**
+ * handle takes the programId parameter and returns
+ * the program associated with the programId
+ * @param req API request object
+ * @param res API response object
+ */
+export default async function handle(
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> {
+    if (req.method == "GET") {
+        // Obtain program id
+        const { id } = req.query;
+        // Obtain program with provided programId
+        const program = await getProgram(id as string);
+
+        if (!program) {
+            ResponseUtil.returnNotFound(
+                res,
+                `Program with id ${id} not found.`,
+            );
+            return;
+        }
+        ResponseUtil.returnOK(res, program);
+        return;
+    } else if (req.method == "DELETE") {
+        // Obtain program id
+        const { id } = req.query;
+        // Delete program with provided programId
+        const program = await deleteProgram(id as string);
+
+        if (!program) {
+            ResponseUtil.returnNotFound(
+                res,
+                `Program with id ${id} not found.`,
+            );
+            return;
+        }
```

Comment:
I wonder if we should return a NOT FOUND for a DELETE request that doesn't exist because that means it's already "deleted". thoughts? @feng-eric

Reply:
Can you explain what you mean by return a NOT FOUND?

Also, I think there's also the case where we delete something that wasn't there in the first place (doesn't have to be already "deleted")